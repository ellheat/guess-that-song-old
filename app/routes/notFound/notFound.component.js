import React, { PureComponent } from 'react';
import Helmet from 'react-helmet';

import { Container } from './notFound.styles';

export class NotFound extends PureComponent {
  render() {
    return (
      <Container>
        <Helmet
          title="Not Found"
        />

        <h1>404</h1>
      </Container>
    );
  }
}
