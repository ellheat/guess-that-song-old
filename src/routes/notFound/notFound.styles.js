import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`;

export const Heading = styled.h1`

`;

export const Button = styled.button`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`;
