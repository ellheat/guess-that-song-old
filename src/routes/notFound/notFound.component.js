import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { injectIntl, FormattedMessage } from 'react-intl';
import { compose } from 'ramda';

import messages from './notFound.messages';
import { Container, Heading, Button } from './notFound.styles';
import { ROUTES } from '../../constants';

export class NotFoundComponent extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
  };

  handleExploreClick = () => {
    this.props.history.push(ROUTES.home);
  };

  render() {
    return (
      <Container>
        <Helmet
          title={this.props.intl.formatMessage(messages.pageTitle)}
        />

        <Heading><FormattedMessage {...messages.title} /></Heading>

        <Button light onClick={this.handleExploreClick}>
          <FormattedMessage {...messages.button} />
        </Button>
      </Container>
    );
  }
}

export const NotFound = compose(
  injectIntl,
)(NotFoundComponent);
