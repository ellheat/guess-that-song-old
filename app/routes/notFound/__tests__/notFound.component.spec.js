import React from 'react';
import { shallow } from 'enzyme';

import { NotFound } from '../notFound.component';


describe('NotFound: Component', () => {
  const component = (props) => (
    <NotFound {...props} />
  );

  it('should not render not-found root', () => {
    const wrapper = shallow(component({}));
    global.expect(wrapper).toMatchSnapshot();
  });
});
