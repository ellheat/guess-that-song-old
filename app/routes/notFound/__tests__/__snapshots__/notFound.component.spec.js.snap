// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound: Component should not render not-found root 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Not Found"
  />
  <h1>
    404
  </h1>
</styled.div>
`;
